wagtail/admin/views/generic/models.py

```diff
@@ -256,13 +256,29 @@ def get_buttons(column, instance, *args, **kwargs):
         )
 
     def _get_custom_column(self, field_name, column_class=Column, **kwargs):
-        label, attr = label_for_field(field_name, self.model, return_attr=True)
+        *relation, field = field_name.split(".")
+        model_class = self.model
+
+        # iterate over the relation list to try to get the last model
+        # where the field exists
+        for model in relation:
+            model_class = getattr(model_class, "_meta").get_field(model).related_model
+            model_name = model
+
+        label, attr = label_for_field(field, model_class, return_attr=True)
+
+        # add the name of relational model to make the label more clear
+        try:
+            label = model_name + " " + label
+        except NameError:
+            pass
+
         sort_key = getattr(attr, "admin_order_field", None)
 
         # attr is None if the field is an actual database field,
         # so it's possible to sort by it
         if attr is None:
-            sort_key = field_name
+            sort_key = field_name.replace(".", "_")
```

Comment:
Does this work? I imagine you'd need double underscores for this to work. If we used `LOOKUP_SEP` instead of dot-separated fields, we probably wouldn't need this.

Reply:
Yes you are totally right, if `LOOKUP_SEP` is used there is no need for this.